jmxtrans-core/src/main/java/com/googlecode/jmxtrans/jmx/JmxResultProcessor.java

```diff
@@ -111,8 +111,16 @@ private void getResult(Builder<Result> accumulator, String attributeName, Compos
 		Map<String, Object> values = newHashMap();
 
 		Set<String> keys = t.keySet();
+		keyLoop:
 		for (String key : keys) {
 			Object value = cds.get(key);
+			if (query.getKeys().size() > 0) {
```

Comment:
The `for` loop will not be entered if `query.getKeys()` is empty, so this `if` is actually unnecessary.